test/e2e/support.js

```diff
@@ -24,7 +24,47 @@ Cypress.Commands.add('placeBidByDiscount', (params) => {
 
   const command = `agops inter bid by-discount --from ${fromAddress} --give ${giveAmount} --discount ${discount} --keyring-backend=test`;
 
-  cy.exec(command, { env: { AGORIC_NET } }).then(({ stdout }) => {
-    expect(stdout).to.contain('Your bid has been accepted');
+  cy.exec(command, { env: { AGORIC_NET }, failOnNonZeroExit: false }).then(
+    ({ stdout }) => {
+      expect(stdout).to.contain('Your bid has been accepted');
+    },
+  );
+});
+
+Cypress.Commands.add('checkAuctionStatus', () => {
+  cy.exec(`agops inter auction status`, {
+    env: { AGORIC_NET },
+    failOnNonZeroExit: false,
+  }).then(({ stdout }) => {
+    const output = JSON.parse(stdout);
+
+    function checkFieldPresence(field) {
+      const fieldValue = Cypress._.get(output, field);
```

Comment:
This fails on the first missing key and doesn't mention the others.

is `Cypress._get` doing anything more than `output[field]`?

If that's just a JSON object or there's a way to get a plain JSON object, you could use Endo patterns. Or `Object.keys().sort().join(',')` and string compare

Reply:
Made another change [here](https://github.com/Agoric/wallet-app/compare/f48f6a4a9ac86a7d204a3693a35be372a4a88e87..cb5f025b24a801c4011db90ba9be923b6a903e79).

Removed string comparison because the JSON returned by `agops inter auction  status` can have extra fields. For example, it failed because of missing `startPrice` over [here](https://github.com/Agoric/wallet-app/actions/runs/9158454163/job/25176854917). I reckon we also have a field `currentPriceLevel`, which can also make it to fail. 

I've revised the implementation a little because the JSON returned by `agops inter auction status` might include additional fields beyond those specified in `requiredFields`. A failure occurred due to the absence of `startPrice` over [here](https://github.com/Agoric/wallet-app/actions/runs/9158454163/job/25176854917)  We're also aware of another potential field, `currentPriceLevel`, which could similarly cause a failure.

Reply:
I've revised the [implementation](https://github.com/Agoric/wallet-app/compare/f48f6a4a9ac86a7d204a3693a35be372a4a88e87..cb5f025b24a801c4011db90ba9be923b6a903e79) a little because the JSON returned by `agops inter auction status` might include additional fields beyond those specified in `requiredFields`. A failure occurred due to the absence of `startPrice` over [here](https://github.com/Agoric/wallet-app/actions/runs/9158454163/job/25176854917)  We're also aware of another potential field, `currentPriceLevel`, which could similarly cause a failure.

Reply:
Awesome...........Thanks a ton for sharing, Turadg! :)